packages/zwave-js/src/lib/commandclass/PowerlevelCC.ts

```diff
@@ -0,0 +1,409 @@
+			case PowerlevelCommand.Get:
+			case PowerlevelCommand.TestNodeGet:
+				return this.isSinglecast();
+			case PowerlevelCommand.Set:
+			case PowerlevelCommand.TestNodeSet:
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	public async setNormalPowerlevel(): Promise<void> {
+		this.assertSupportsCommand(PowerlevelCommand, PowerlevelCommand.Set);
+
+		const cc = new PowerlevelCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			powerlevel: Powerlevel.NormalPower,
+		});
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async setCustomPowerlevel(
+		powerlevel: Powerlevel,
+		timeout: number,
+	): Promise<void> {
+		this.assertSupportsCommand(PowerlevelCommand, PowerlevelCommand.Set);
+
+		const cc = new PowerlevelCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			powerlevel,
+			timeout,
+		});
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async getPowerlevel(): Promise<
+		Pick<PowerlevelCCReport, "powerlevel" | "timeout"> | undefined
+	> {
+		this.assertSupportsCommand(PowerlevelCommand, PowerlevelCommand.Get);
+
+		const cc = new PowerlevelCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = await this.driver.sendCommand<PowerlevelCCReport>(
+			cc,
+			this.commandOptions,
+		);
+		if (response) {
+			return pick(response, ["powerlevel", "timeout"]);
+		}
+	}
+
+	public async startTestNode(
+		testNodeId: number,
+		powerlevel: Powerlevel,
+		testFrameCount: number,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			PowerlevelCommand,
+			PowerlevelCommand.TestNodeSet,
+		);
+
+		const cc = new PowerlevelCCTestNodeSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			testNodeId,
+			powerlevel,
+			testFrameCount,
+		});
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async getTestNodeResult(): Promise<
+		| Pick<
+				PowerlevelCCTestNodeReport,
+				"testNodeId" | "testStatus" | "testFrameAcknowlegedCount"
+		  >
+		| undefined
+	> {
+		this.assertSupportsCommand(
+			PowerlevelCommand,
+			PowerlevelCommand.TestNodeGet,
+		);
+
+		const cc = new PowerlevelCCTestNodeGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = await this.driver.sendCommand<PowerlevelCCTestNodeReport>(
+			cc,
+			this.commandOptions,
+		);
+		if (response) {
+			return pick(response, [
+				"testNodeId",
+				"testStatus",
+				"testFrameAcknowlegedCount",
+			]);
+		}
+	}
+}
+
+@commandClass(CommandClasses.Powerlevel)
+@implementedVersion(1)
+export class PowerlevelCC extends CommandClass {
+	declare ccCommand: PowerlevelCommand;
+}
+
+type PowerlevelCCSetOptions = CCCommandOptions &
+	(
+		| {
+				powerlevel: Powerlevel;
+				timeout: number;
+		  }
+		| {
+				powerlevel: typeof Powerlevel.NormalPower;
+		  }
```

Comment:
This might need `timeout?: undefined` in order to make some checks less awkward in TypeScript. For example the one in the constructor below.